pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -137,7 +140,10 @@ func ipSecReplaceStateOut(remoteIP, localIP net.IP) (uint8, error) {
 		Value: ((spiWide << 12) | linux_defaults.RouteMarkEncrypt),
 		Mask:  linux_defaults.IPsecMarkMask,
 	}
-	state.OutputMark = linux_defaults.RouteMarkEncrypt
+	state.OutputMark = &netlink.XfrmMark{
+		Value: linux_defaults.RouteMarkEncrypt,
+		Mask:  linux_defaults.IPsecMarkMaskIn,
```

Comment:
In ipSecReplaceStateOut I believe we want linux_defaults.IPsecMarkMask not the In version of it.

On out we have skb mark like this, 'skb->mark = 0xIDIDFFID' where the 0xID is the 3 bytes of security context split across the mark. And the 0XFF is the encrypt key used to match with IPsec SPI value. So double check please with a test, but I believe its best to drop the "In" above.

Reply:
Above is wrong (I left though to remind myself I already consider it).

Actually we are just setting the RouteMarkEncrypt=0x0E00 and RouteMarkDecrypt=0x0D00 values. So IPsecMarkMaskIn=0x0F00 is correct, but change to RouteMarkMask so that it reads better and doesn't confuse us later. This is a bit subtle so best to be clear if possible